go/vt/vtgate/planbuilder/set.go

```diff
@@ -222,14 +211,39 @@ func buildSetPlan(stmt *sqlparser.Set, vschema ContextVSchema) (engine.Primitive
 	}, nil
 }
 
-func planTabletInput(vschema ContextVSchema, tabletExpressions []*sqlparser.SetExpr) (engine.Primitive, error) {
+type expressionConverter struct {
+	tabletExpressions []*sqlparser.SetExpr
+}
+
+func (spb *expressionConverter) convert(setExpr *sqlparser.SetExpr) (evalengine.Expr, error) {
+	astExpr := setExpr.Expr
+	evalExpr, err := sqlparser.Convert(astExpr)
+	if err != nil {
+		if err != sqlparser.ExprNotSupported {
+			return nil, err
+		}
+		// We have an expression that we can't handle at the vtgate level
+		if !expressionOkToDelegateToTablet(astExpr) {
+			// Uh-oh - this expression is not even safe to delegate to the tablet. Give up.
+			return nil, vterrors.Errorf(vtrpcpb.Code_UNIMPLEMENTED, "expression not supported for SET: %s", sqlparser.String(astExpr))
```

Comment:
If we can parse the expression than we can recursively call planbuilder and put that as input for this setOp.

Reply:
I'm not changing semantics or behaviour in this PR, just moving code around to make it cleaner. If you want to change this, we should do that in a separate PR.